### PR TITLE
add set-marketplace rpc

### LINKED ISSUE #950: Add ability to change Registry
At minimum, it needs to be possible to change registry for alpha/beta testing, but it's also a big thesis of the company that the ability of someone else to host a Marketplace at least be available

Reported by: @kn0wmad

### PR DESCRIPTION
**Pull request by dr-bonez**
closes #950

### CODE REVIEW

**Comment by kn0wmad** [comment]
Awesome, thank you!  Can you roughly document the process to set?

**Comment by dr-bonez** [comment]
`embassy-cli server set-marketplace <url>`
or
`embassy-cli package set-marketplace <url>` to only change the marketplace for fetching packages, but not os updates.

**Comment by dr-bonez** [comment]
note: this cli option is not very defensive. if you put in an invalid url, it can break your embassy.

**Comment by chrisguida** [comment]
Can I fix it by calling it again with a valid url?

**Comment by kn0wmad** [comment]
Nice that you can exclude OS updates.  When you say "break," how bad is that?  Just cannot access Marketplace?

File: appmgr/src/context/rpc.rs [hunk 1]
```diff
@@ -248,6 +249,38 @@ impl RpcContext {
...
+    #[instrument(skip(self, db))]
+    pub async fn set_nginx_conf<Db: DbHandle>(&self, db: &mut Db) -> Result<(), Error> {
+        tokio::fs::write("/etc/nginx/sites-available/default", {
+            let info = crate::db::DatabaseModel::new()
+                .server_info()
+                .get(db, true)
+                .await?;
+            format!(
+                include_str!("../nginx/main-ui.conf.template"),
+                lan_hostname = info.lan_address.host_str().unwrap(),
```
Comment (Blu-J) [inline appmgr/src/context/rpc.rs:261]: Thanks for using a named variables for this template format.

**Comment by Blu-J** [comment]
Code Looks Good to me

**Comment by dr-bonez** [comment]
> When you say "break," how bad is that? Just cannot access Marketplace?

Nginx will die. Technically it won't commit the change if this happens, so if you restart the embassy it will be fine.

**Pull request merged**